src/hotspot/share/memory/memoryReserver.cpp

```diff
@@ -575,38 +575,41 @@ ReservedHeapSpace HeapReserver::Instance::reserve_compressed_oops_heap(const siz
 
     // Attempt to allocate so that we can run without base and scale (32-Bit unscaled compressed oops).
     // Give it several tries from top of range to bottom.
-    if (aligned_heap_base_min_address + size <= (char *)UnscaledOopHeapMax) {
+    if (aligned_heap_base_min_address + size <= UnscaledOopHeapMax) {
 
       // Calc address range within we try to attach (range of possible start addresses).
-      char* const highest_start = align_down((char *)UnscaledOopHeapMax - size, attach_point_alignment);
-      char* const lowest_start  = align_up(aligned_heap_base_min_address, attach_point_alignment);
-      reserved = try_reserve_range(highest_start, lowest_start, attach_point_alignment,
-                                   aligned_heap_base_min_address, (char *)UnscaledOopHeapMax, size, alignment, page_size);
+      uintptr_t const highest_start = align_down(UnscaledOopHeapMax - size, attach_point_alignment);
+      uintptr_t const lowest_start  = align_up(aligned_heap_base_min_address, attach_point_alignment);
+      assert(lowest_start <= highest_start, "lowest: " INTPTR_FORMAT " highest: " INTPTR_FORMAT ,
+                                          lowest_start, highest_start);
+      reserved = try_reserve_range((char*)highest_start, (char*)lowest_start, attach_point_alignment,
+                                   (char*)aligned_heap_base_min_address, (char*)UnscaledOopHeapMax, size, alignment, page_size);
     }
 
     // zerobased: Attempt to allocate in the lower 32G.
-    char *zerobased_max = (char *)OopEncodingHeapMax;
+    size_t zerobased_max = OopEncodingHeapMax;
```

Comment:
Again not obvious what this improves. We obviously have very inconsistent use of types here in that we loosely use `char*`, `uint64_t` and `size_t` to all mean a 64-bit unsigned value, ansd no matter what types we use in the declarations we have to cast something somewhere.

Reply:
For your suggested changes, I need them be more precise and explicit to be able to implement. TIA